pkg/endpointmanager/idallocator/allocator.go

```diff
@@ -26,17 +27,22 @@ const (
 )
 
 var (
-	pool = idpool.NewIDPool(minID, maxID)
+	mutex = lock.RWMutex{} // Protects access to `pool` below.
+	pool  = idpool.NewIDPool(minID, maxID)
```

Comment:
This will slow down the access of the `pool` which already has a mutex builtin. I know the mutex is protecting the variable `pool` not the contents of `pool`. This entire mutex is only being added because of `ReallocatePool` which is only used in unit tests. Since this is the case we should avoid using a mutex entirely and re-write the `ReallocatePool` function to do something like:
```
// ReallocatePool starts over with a new pool.
// This function is only used for tests and its implementation is not
// optimized for production.
func ReallocatePool() {
	for i := minID; i < maxID; i++ {
		pool.Release(i)
	}
}
```

Reply:
Ah, thanks for extra insight. I agree that the mutex would slow things down. Good catch on noticing that `ReallocatePool` isn't used in implementation code. 

Will add this and remote the mutex.